src/main/java/baritone/utils/schematic/Schematic.java

```diff
@@ -18,53 +18,72 @@
-        byte[] blocks = schematic.getByteArray("Blocks");
-        byte[] metadata = schematic.getByteArray("Data");
 
-        byte[] additional = null;
-        if (schematic.hasKey("AddBlocks")) {
-            byte[] addBlocks = schematic.getByteArray("AddBlocks");
-            additional = new byte[addBlocks.length * 2];
-            for (int i = 0; i < addBlocks.length; i++) {
-                additional[i * 2 + 0] = (byte) ((addBlocks[i] >> 4) & 0xF); // lower nibble
-                additional[i * 2 + 1] = (byte) ((addBlocks[i] >> 0) & 0xF); // upper nibble
-            }
-        }
+        CompoundNBT regions = schematic.getCompound("Regions");
+        CompoundNBT metadata = schematic.getCompound("Metadata");
+        widthX = metadata.getCompound("EnclosingSize").getInt("x");
+        heightY = metadata.getCompound("EnclosingSize").getInt("y");
+        lengthZ = metadata.getCompound("EnclosingSize").getInt("z");
         states = new BlockState[widthX][lengthZ][heightY];
-        for (int y = 0; y < heightY; y++) {
-            for (int z = 0; z < lengthZ; z++) {
-                for (int x = 0; x < widthX; x++) {
-                    int blockInd = (y * lengthZ + z) * widthX + x;
 
-                    int blockID = blocks[blockInd] & 0xFF;
-                    if (additional != null) {
-                        // additional is 0 through 15 inclusive since it's & 0xF above
-                        blockID |= additional[blockInd] << 8;
+        for (String regionKey : regions.keySet()) {
+            CompoundNBT region = regions.getCompound(regionKey);
+            // BlockPos regionPos = NBTUtil.readBlockPos(region.getCompound("Position")); Not doing this because this method expects uppercase "X", "Y", "Z" tag keys in the compound. In litematic those are lowercase.
+            int posX = region.getCompound("Position").getInt("x");
+            int posY = region.getCompound("Position").getInt("y");
+            int posZ = region.getCompound("Position").getInt("z");
+            BlockPos regionPos = new BlockPos(posX, posY, posZ);
+            // BlockPos regionSize = NBTUtil.readBlockPos(region.getCompound("Size")); Same goes for region size.
+            int sizX = region.getCompound("Size").getInt("x");
+            int sizY = region.getCompound("Size").getInt("y");
+            int sizZ = region.getCompound("Size").getInt("z");
+            BlockPos regionSize = new BlockPos(Math.abs(sizX), Math.abs(sizY), Math.abs(sizZ));
+            ListNBT blockStatePalette = region.getList("BlockStatePalette", 10);
+            long[] blockStateArr = region.getLongArray("BlockStates");
+
+
+            BlockPos posEndRel = PositionUtils.getRelativeEndPositionFromAreaSize(new BlockPos(sizX, sizY, sizZ)).add(regionPos);
+            BlockPos posMin = PositionUtils.getMinCorner(regionPos, posEndRel);
+            LitematicaBlockStateContainer container = LitematicaBlockStateContainer.createFrom(blockStatePalette, blockStateArr, regionSize);
```

Comment:
Forgot to mention but is it really necessary to even use a custom implementation here?